dashboard/app/models/user.rb

```diff
@@ -2764,6 +2766,26 @@ def should_validate_us_state?
     new_record? || us_state_changed?
   end
 
+  def student_in_lockout_flow?
+    return unless user_type == TYPE_STUDENT
+    return unless %w[US RD].include? country_code
+    return unless birthday_changed? || us_state_changed?
+
+    # Create copy of user to mock the user's state before an update.
+    user_before_update = dup
+
+    # If us_state or age is changed, set to previous value before the update to be used to check if the user is in the lockout flow.
+    user_before_update.us_state = properties_was['us_state'] if us_state_changed?
+    user_before_update.birthday = birthday_was if birthday_changed?
```

Comment:
```ruby
user_before_update.attributes = {
  us_state: property_before_save('us_state'),
  birthday: birthday_was
}
```

Or, I haven't quite tested this interaction, but I think this works, just:

```ruby
user_before_update = User.new(attributes.deep_merge(changed_attributes))
```

Reply:
Ill give it a try Thanks Wilkie!